Game.Checkers/Checkers/DrawBoard.cs

```diff
@@ -0,0 +1,221 @@
+            if (piece == Piece.WhitePawn)
+            {
+                DrawPawn(context, white, grey);
+                return;
+            }
+            if (piece == Piece.BlackPawn)
+            {
+                DrawPawn(context, black, grey);
+                return;
+            }
+            if (piece == Piece.WhiteCrowned)
+            {
+                DrawCrowned(context, white, grey);
+                return;
+            }
+            if (piece == Piece.BlackCrowned)
+            {
+                DrawCrowned(context, black, grey);
+                return;
+            }
+        }
+
+        private void DrawPawn(Context context, Color fill, Color border)
+        {
+            double lineWidth = context.LineWidth;
+            context.LineWidth = 0.03;
+
+            context.Arc(0.5, 0.5, 0.3, 0, 2 * Math.PI);
+            context.SetSourceColor(fill);
+            context.FillPreserve();
+
+            context.SetSourceColor(border);
+            context.Stroke();
+
+            context.Arc(0.5, 0.5, 0.15, 0, 2 * Math.PI);
+            context.Stroke();
+
+            context.LineWidth = lineWidth;
+        }
+
+        private void DrawCrowned(Context context, Color fill, Color border)
+        {
+            double lineWidth = context.LineWidth;
+            context.LineWidth = 0.03;
+            context.Arc(0.5, 0.5, 0.3, 0, 2 * Math.PI);
+            context.SetSourceColor(fill);
+            context.FillPreserve();
+
+            context.SetSourceColor(border);
+            context.Stroke();
+
+            context.Arc(0.5, 0.5, 0.15, 0, 2 * Math.PI);
+            context.SetSourceRGB(0.831, 0.686, 0.216);
+            context.FillPreserve();
+
+            context.SetSourceColor(border);
+            context.Stroke();
+
+            context.LineWidth = lineWidth;
+        }
+
+        private void DrawGameState(Context context, ICheckersInputState inputState, CheckersState state, IEnumerable<(CheckersAction, double)> ratedActions)
+        {
+            if (inputState is DefaultCIS) // draw best actions
+            {
+                Color green = new(0.5, 1, 0);
+                foreach (var action in ratedActions.Reverse())
+                {
+                    DrawRatedAction(context, state, action, green);
+                }
+                return;
+            }
+            if (inputState is MarkedPieceCIS markedPieceState) // draw actions for marked piece
+            {
+                Color blue = new(0, 0.75, 1);
+                foreach (var action in ratedActions)
+                {
+                    DrawRatedAction(context, state, action, blue);
+                }
+                DrawMarkedField(context, state, markedPieceState.MarkedField);
+                return;
+            }
+            { // draw actions for ongoing action
+                Color blue = new(0, 0.75, 1);
+                CaptureActionInProgressCIS actionInProgressState = (CaptureActionInProgressCIS)inputState;
```

Comment:
czemu służy to rzutowanie?

Reply:
No jak sprawdziłem pozostałe przypadki to zostaje tylko opcja, że inputState jest tego właśnie typu. Muszę zrzutować żeby się dostać do np. VisitedFields